### ISSUE TITLE
Linux Use

### ISSUE THREAD

**Issue by AaronBeisaw** [briansd9/exile-diary#112]
Hi, I'm trying to use on Manjaro and using wine the app starts but none of the text boxes for Account, POSESSID, etc are allowing me to type. I am able to browse for client.txt but I'm not sure if that is saving the location either as I can not see it. 
![EDiary](https://user-images.githubusercontent.com/54881462/111052291-a0b8f900-8427-11eb-83f3-7b273869b731.png)

**Comment by briansd9** [comment]
Hmm, Linux not officially supported as I don't use it myself, many things may be broken... but let's give it a try

You can create the settings file manually - fill up [this samplesettings.txt](https://github.com/briansd9/exile-diary/files/6135914/samplesettings.txt) with your account info, then save it to `settings.json` in the app's data directory (`%appdata%\exile-diary\` for me, not sure about yours)

Also, can you press `Ctrl+Shift+1`? A developer console should pop up on the right side of the screen, can you send me any errors that appear there?

**Comment by AaronBeisaw** [comment]
Ill try manually saving the settings.json file, though the developer console seems to freeze when opened. I'll update on that if I manage to get it working. Thank you!

**Comment by AaronBeisaw** [comment]
Manually saving the settings.json worked to get it reading my character but it cant seem to see the client.txt, and I can not use any of the buttons on the left hand side (e.g. trying to manually get to the settings page), and the dev console still completely freezes up - can't even hit the close button on it.

**Comment by AaronBeisaw** [comment]
This error just popped upon exiting a map
![EDiaryError1](https://user-images.githubusercontent.com/54881462/111055629-074d0f80-8446-11eb-8ef8-63cd8118e26a.png)

**Comment by briansd9** [comment]
Hmm, did you edit or delete your Client.txt file while the app was open? That might have caused the error.

Can you send me your log file (`log.txt` located in the app data directory)? Will take a look.

**Comment by AaronBeisaw** [comment]
I May have cleared client.txt while it was open, though this persists even after restarting. Here is log.txt and log1.txt, wasnt sure if you needed both.

[log.txt](https://github.com/briansd9/exile-diary/files/6136186/log.txt)
[log1.txt](https://github.com/briansd9/exile-diary/files/6136187/log1.txt)

**Comment by briansd9** [comment]
Does the "bad file descriptor" error window pop up consistently?

From reading the log file, it looks like Exile Diary is temporarily banned from connecting to the PoE servers (when it tried putting in all your old map runs, it also attempted to check your character's inventory several hundred times)... not sure how long until the ban expires, but try again later.

**Comment by AaronBeisaw** [comment]
Yes that error started popping up consistently before, though I did not see it until after the chat error. Either way, I'll try again later and update back here again. Thanks!

**Comment by AaronBeisaw** [comment]
12 hours later, same JavaScript and area change errors

**Comment by briansd9** [comment]
Hmm, I must admit I'm out of ideas, sorry... I've filed a bug report for the module I'm using, hopefully the developer will be able to help.

**Comment by AaronBeisaw** [comment]
No worries! I know linux is not officially supported, and I appreciate your help. If theres anything else I can do to help just let me know.

**Comment by Anupalabdhi** [comment]
Hey,

Sorry for posting on an old issue, but I just realized that "Exile-Diary" compiles quite nicely for linux, and by using the app as a native application, I was able to bypass quite a lot of the issues mentioned by the initial issue poster. Since this issue has been kept open and the topic is for linux use, I thought I might as well post here.

The app seems to be working nicely in my initial testing. I compiled an app-image for my personal use as that is what I prefer for such apps. @briansd9, if you don't mind, you can look into posting "unsupported" instructions for making a linux build for people like Aaron, so that at least they won't have to finagle with Wine. Interestingly enough I found this app because I was unable to cleanly build Exilence. Let me know if you need additional info or logs too.

**Comment by AaronBeisaw** [comment]
@Anupalabdhi Any chance you could comment on how to compile an AppImage for exile-diary? I'm familiar with linux but still rather new, this is something I haven't had to do yet

**Comment by Anupalabdhi** [comment]
Sure @AaronBeisaw , For posterity's sake, I'll list out the steps that I used (adapted from [this](https://github.com/klayveR/xenontrade/wiki/Build-it-yourself)) :-

1. Use `git clone https://github.com/briansd9/exile-diary.git` to create a copy of the source code in a folder named "exile-diary" in the place where you executed the above command, usually your home directory.
2. `cd exile-diary` to move to the directory
3. Use a text editor of your preference to edit the package.json file. Specifically line 25 of that file which states  `"target": "deb"` Replace the `deb` with `AppImage`. Save the file.
4. `npm install`
5. `npm run dist`
6. It will create a folder named `Dist` which should contain your appimage file. You can just use that to run the program which will create it's config and cache in `~/.config/exile-diary`.

Hope this helps. Feel free to comment if you get stuck at any point.

**Comment by AaronBeisaw** [comment]
@Anupalabdhi worked like a charm, thanks so much!
